File: 0443-string-compression/0443-string-compression.cpp
// ln->length of group.
// st->used for string representation of 'ln'.

class Solution {
public:
    int compress(vector<char>& chars) {
        int i=0,res=0;
        while(i<chars.size()){
            int ln = 1;
            while(i+ln<chars.size() && chars[i+ln]==chars[i])
                ln++;
            chars[res++] = chars[i];
            if(ln>1){
                for(char c:to_string(ln)){
                    chars[res++] = c;
                }
            }
            i+=ln;
        }
        return res;
    }
};

File: 0443-string-compression/0443-string-compression.py
#ln->length of group.
#st->used for string representation of 'ln'.

class Solution:
    def compress(self, chars: List[str]) -> int:
        n = len(chars)
        i,res = 0,0
        while i < n:
            ln = 1
            while (i+ln<len(chars) and chars[i+ln]==chars[i]):
                ln += 1
            chars[res] = chars[i]
            res+=1
            if(ln>1):
                st = str(ln)
                chars[res:res+len(st)] = list(st)
                res+=len(st)
            i+=ln
        return res